spec/services/calculators/dependant_allowance_calculator_spec.rb

```diff
@@ -0,0 +1,85 @@
+require 'rails_helper'
+
+module Calculators
+  RSpec.describe DependantAllowanceCalculator do
+    describe '#call' do
+      subject { described_class.new(dependant).call }
+      before do
+        allow_any_instance_of(described_class).to receive(:child_under_15_allowance).and_return 111.11
+        allow_any_instance_of(described_class).to receive(:child_aged_15_allowance).and_return 222.22
+        allow_any_instance_of(described_class).to receive(:child_16_and_over_allowance).and_return 333.33
+        allow_any_instance_of(described_class).to receive(:adult_allowance).and_return 444.44
+      end
+
+      context 'under 15 with income' do
+        let(:dependant) { create :dependant, :under_15, monthly_income: 25.00 }
+        it 'returns the child under 15 allowance and does not subtract the income' do
+          expect(subject).to eq 111.11
+        end
+      end
+
+      context 'under 15 without income' do
+        let(:dependant) { create :dependant, :under_15, monthly_income: 0.0 }
+        it 'returns the child under 15 allowance' do
+          expect(subject).to eq 111.11
+        end
+      end
+
+      context '15 years old with income' do
+        let(:dependant) { create :dependant, :aged_15, monthly_income: 25.50 }
+        it 'returns the aged 15 allowance less the monthly income' do
+          expect(subject).to eq(222.22 - 25.50)
+        end
+      end
+
+      context '15 years old without income' do
+        let(:dependant) { create :dependant, :aged_15, monthly_income: 0.0 }
+        it 'returns the aged 15 allowance less the monthly income' do
+          expect(subject).to eq 222.22
+        end
+      end
+
+      context '16 or 17 years old in full time education with no income' do
+        let(:dependant) { create :dependant, :aged_16_or_17, monthly_income: 0.0, in_full_time_education: true }
+        it 'returns the child 16 or over with no income deduction' do
+          expect(subject).to eq 333.33
+        end
+      end
+
+      context '16 years old in full time education with income' do
```

Comment:
Should we test that `full_time_education = false` returns `adult_allowance - @dependant.monthly_income`?